lib/rspec/core/profiler.rb

```diff
@@ -0,0 +1,29 @@
+module RSpec
+  module Core
+    # @private
+    class Profiler
```

Comment:
Given that this really only provides profiling for example groups (since each individual example profiles itself), do you think it should be called `ExampleGroupProfiler`?  That would also help differentiate it from the `ProfileFormatter`.

On a side note, I think I'd prefer to see this in the `formatters` directory.  No, it's not an object that formats output, but it gets registered as a formatter with the reporter and that's where we generally put those classes.  Plus it's nice not to increase the file count in the already crowded `rspec/core` dir.